src/main/java/com/ms3_inc/camel/extensions/rest/SwaggerRequestValidator.java

```diff
@@ -16,91 +16,143 @@
  * limitations under the License.
  */
 
-import com.atlassian.oai.validator.OpenApiInteractionValidator;
-import com.atlassian.oai.validator.model.Request;
-import com.atlassian.oai.validator.model.SimpleRequest;
-import com.atlassian.oai.validator.report.SimpleValidationReportFormat;
-import com.atlassian.oai.validator.report.ValidationReport;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.dataformat.xml.XmlMapper;
 import com.ms3_inc.camel.extensions.rest.exception.BadRequestException;
 import org.apache.camel.AsyncCallback;
-import org.apache.camel.CamelException;
 import org.apache.camel.Exchange;
 import org.apache.camel.support.AsyncProcessorSupport;
 import org.apache.camel.support.MessageHelper;
+import org.openapi4j.core.exception.ResolutionException;
+import org.openapi4j.core.validation.ValidationException;
+import org.openapi4j.core.validation.ValidationResults;
+import org.openapi4j.operation.validator.model.Request;
+import org.openapi4j.operation.validator.model.impl.Body;
+import org.openapi4j.operation.validator.model.impl.DefaultRequest;
+import org.openapi4j.operation.validator.validation.RequestValidator;
+import org.openapi4j.parser.OpenApi3Parser;
+import org.openapi4j.parser.model.v3.OpenApi3;
+import org.openapi4j.parser.model.v3.Server;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import org.springframework.util.MultiValueMap;
-import org.springframework.web.util.UriComponentsBuilder;
+import org.springframework.core.io.ClassPathResource;
 
+import java.io.IOException;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Map;
 
 public class SwaggerRequestValidator extends AsyncProcessorSupport {
-    private final Logger LOGGER = LoggerFactory.getLogger(getClass());
-    private final OpenApiInteractionValidator validator;
+    private final Logger LOGGER = LoggerFactory.getLogger(SwaggerRequestValidator.class);
+    private final RequestValidator openapi4jValidator;
+    private OpenApi3 api;
 
     public SwaggerRequestValidator(String specPath) {
-        validator = OpenApiInteractionValidator
-                .createFor(specPath)
-                .build();
+        api = null;
```

Comment:
just forward the parameters to the other constructors: `this(specPath, null)` and in the other constructor conditionally make use of the basePath if not null